activerecord/test/cases/timestamp_test.rb

```diff
@@ -457,6 +464,14 @@ def test_index_is_created_for_both_timestamps
   ensure
     ActiveRecord::Base.connection.drop_table(:foos)
   end
+
+  def test_datetime_precision_applied_to_timestamp
+    d = Parrot.create!(name: 'Bogdan')
+    in_instance = d.created_at
+    from_db = Parrot.find(d.id).created_at
+    assert_equal from_db.usec, in_instance.usec
+    assert_equal from_db, in_instance
```

Comment:
I don't see any test like that inside `am/test/cases/type/` that uses `Model`. They all just instantiate the `AM::Type::*` object and call `#cast` method there.

Also, testing this feature in AM will not cover this change:
https://github.com/rails/rails/pull/20317/files#diff-2695559f74d901580d9c9314a9f4aff7R14
that caused postgresql specific bug.

I would stick with test we have know. I will only change it with  use of `#reload`.